docs/strategy-advanced.md

```diff
@@ -42,32 +42,34 @@ class AwesomeStrategy(IStrategy):
 
 ***
 
-### Storing custom information using DatetimeIndex from `dataframe`
+## Custom sell signal
 
-Imagine you need to store an indicator like `ATR` or `RSI` into `custom_info`. To use this in a meaningful way, you will not only need the raw data of the indicator, but probably also need to keep the right timestamps.
+It is possible to define custom sell signals. This is very useful when we need to customize sell conditions for each individual trade.
 
-```python
-import talib.abstract as ta
-class AwesomeStrategy(IStrategy):
-    # Create custom dictionary
-    custom_info = {}
+An example of how we can set stop-loss and take-profit targets in the dataframe and also sell trades that were open longer than 1 day:
 
-    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
-        # using "ATR" here as example
-        dataframe['atr'] = ta.ATR(dataframe)
-        if self.dp.runmode.value in ('backtest', 'hyperopt'):
-          # add indicator mapped to correct DatetimeIndex to custom_info
-          self.custom_info[metadata['pair']] = dataframe[['date', 'atr']].set_index('date')
-        return dataframe
+``` python
+class AwesomeStrategy(IStrategy):
+    def custom_sell(self, pair: str, trade: Trade, current_time: datetime, current_rate: float,
+                    current_profit: float, dataframe: Dataframe, **kwargs) -> bool:
+        trade_row = dataframe.loc[dataframe['date'] == timeframe_to_prev_date(trade.open_date_utc)].squeeze()
```

Comment:
What do you mean it cant happen due to last condition? You just said it can happen 👀 I did not think about old-enough trades.. Which makes me wonder if this is a robust solution at all. Old way (copying columns) also suffered from this but.. Be nice if we could 100% reliably access candle data around trades.

Reply:
> What do you mean it cant happen due to last condition

at the end of this function, you're selling if the pair is > 1 day - so it's unlikely that you'll not have enough data for this example if a user copies the whole function (Which i would not rely on ...).

> Which makes me wonder if this is a robust solution at all

guess why i am not a huge fan of this ... as there's a lot of "ifs" that you need to consider.

Using "more" data can be problematic - assume i use a slow trading strategy which keeps trades open for weeks (ignore timeframe for now ...)
The current "live" lookback is 1 call to the exchange's ohlcv / klines endpoint. going back 1 week on 5m candles would mean 2-3 calls - which will slow the data gathering down (by quite some time, if you need to do this for 20 open trades and 5 weeks old trades - and potentially result in DDos problems (considering that binance seems to tighten these limits lately ...) - which obviously can be resolved by using a "slower" ratelimit...

Reply:
> at the end of this function, you're selling if the pair is > 1 day - so it's unlikely that you'll not have enough data for this example if a user copies the whole function (Which i would not rely on ...).

It still can happen though. For example if bot was offline for a longer period of time and then was brought back online this would throw an exception for any unsold trades.

> Using "more" data can be problematic

I can hardly imagine where we would need more data than some candles around buy signal or trade open. It would probably cover 99% of usecases if data around those times is available, even if there is a hole between that time and latest candles.